modules/core/src/main/java/org/apache/ignite/internal/processors/dr/IgniteDrDataStreamerCacheUpdater.java

```diff
@@ -86,10 +92,15 @@ public class IgniteDrDataStreamerCacheUpdater implements StreamReceiver<KeyCache
                     new GridCacheDrExpirationInfo(cacheVal, entry.version(), entry.ttl(), entry.expireTime()) :
                     new GridCacheDrInfo(cacheVal, entry.version()) : null;
 
-                if (val == null)
+                if (val == null) {
                     cache.removeAllConflict(Collections.singletonMap(key, entry.version()));
-                else
+                } else {
                     cache.putAllConflict(Collections.singletonMap(key, val));
+
+                    if (log.isTraceEnabled()) {
+                        log.trace("PutAllConflict invoked for the key = " + key);
+                    }
```

Comment:
I'd log this inside the cache.put/removeAllConflict methods.
Also sensitive logging should be enabled for key.

Reply:
> GridToStringBuilder.SensitiveDataLogging sensitiveDataLogging = S.getSensitiveDataLogging();
>              if (sensitiveDataLogging == PLAIN)
>                  keyValue = key.toString();
>              else if (sensitiveDataLogging == HASH)
>                  keyValue = String.valueOf(IgniteUtils.hash(key.toString));